src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,26 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+interface Props {
+  chosenTodo: Todo;
+  onChoosingTodo: (todo: Todo | null) => void;
+}
+
+export const TodoModal: React.FC<Props> = ({ chosenTodo, onChoosingTodo }) => {
+  const [todoOwner, setTodoOwner] = useState<User | null>(null);
+
+  useEffect(() => {
+    getUser(chosenTodo.userId).then(setTodoOwner);
+  }, [chosenTodo]);
```

Comment:
You are initializing your state with props. There are some potential pitfalls with this approach, as it can lead to bugs and unnecessary re-renders. I recommend checking out these articles to understand why it can be problematic: [here](https://stackoverflow.com/a/50403930), [here](https://reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state), and [here](https://reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state).